daemons/gptp/common/ieee1588clock.cpp

```diff
@@ -87,7 +89,32 @@ IEEE1588Clock::IEEE1588Clock
 
 	number_ports = 0;
 
-	this->forceOrdinarySlave = forceOrdinarySlave;
+	if (!automotiveProfileConfig.automotiveProfile) {
+		//The following boolean options are only configurable when automotive profile is enabled
+		automotiveProfileConfig.transmitAnnounce = true;
+		automotiveProfileConfig.forceAsCapable = false;
+		automotiveProfileConfig.negotiateSyncRate = false;
+		automotiveProfileConfig.automotiveState = false;
+		automotiveProfileConfig.automotiveTestMode = false;
+	} else {
+		//external port configuration feature is enabled by default in automotive profile
+		if (extPortConfig.externalPortConfiguration == EXT_DISABLED) {
+			extPortConfig.externalPortConfiguration = EXT_ENABLED;
```

Comment:
Since this piece of code is in the constructor, I add an error log and a status log to notify the user that the static port state is not specified and is set to PTP_SLAVE by default. (I choose PTP_SLAVE because "isGm" is by default false in the original 17.5 code)